app/controllers/cfp/events_controller.rb

```diff
@@ -100,23 +100,26 @@ def confirm
 
   def join
     @token = params[:token] || ''
+    @join_as = params[:join_as] || ''
     @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
 
+    return unless request.post?
+
     deadline = @event&.conference&.call_for_participation&.hard_deadline
     if deadline && Date.today > deadline
-      return redirect_to cfp_root_path, flash: { error: t('cfp.hard_deadline_over') }
+      unless @join_as == 'assistant'
+        return redirect_to cfp_root_path, flash: { error: t('cfp.hard_deadline_over') }
+      end
     end
 
-    return unless request.post?
-
     if @event.nil?
         return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) }
     end
 
     if @event.people.exists?(@person.id)
       redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_token_already_used')
     else
-      @event.event_people << EventPerson.new(person: @person, event_role: 'speaker')
+      @event.event_people << EventPerson.new(person: @person, event_role: @join_as)
```

Comment:
… and here you just pass the event role from what you got from the request above? What stops you making yourself coordinator or submitter?

This doesn't look right.

Reply:
@erdgeist you are right of course. Thanks for spending time on these reviews..

I googled it and limiting parameters to specific value sets does not seem to be something that "strong parameters" deal with. So I ended up adding [this](https://github.com/eladeyal-intel/frab/blob/4c882f10047bb978db0fe35319fbf066653da471/app/controllers/cfp/events_controller.rb#L108):

```ruby
raise Pundit::NotAuthorizedError unless @join_as.in? ['assistant', 'speaker']
```

Basically anyone with a frab login can join to any event as a `submitter` or an `assistant` (and now: nothing else) ; all she needs is the invitation token. `submitter` can joins only during the CfP; `assistants` can join later too.